ProcessLib/SmallDeformation/CreateSmallDeformationProcess.h

```diff
@@ -0,0 +1,108 @@
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESS_LIB_CREATESMALLDEFORMATIONPROCESS_H_
+#define PROCESS_LIB_CREATESMALLDEFORMATIONPROCESS_H_
+
+#include <cassert>
+
+#include "MaterialLib/SolidModels/createLinearElasticIsotropic.h"
+#include "ProcessLib/Utils/ParseSecondaryVariables.h"
+
+#include "SmallDeformationProcessData.h"
+#include "SmallDeformationProcess.h"
+
+namespace ProcessLib
+{
+namespace SmallDeformation
+{
+template <int DisplacementDim>
+class SmallDeformationProcess;
+
+extern template class SmallDeformationProcess<2>;
+extern template class SmallDeformationProcess<3>;
+
+template <int DisplacementDim>
+std::unique_ptr<SmallDeformationProcess<DisplacementDim>>
+createSmallDeformationProcess(
+    MeshLib::Mesh& mesh,
+    typename Process::NonlinearSolver& nonlinear_solver,
+    std::unique_ptr<typename Process::TimeDiscretization>&& time_discretization,
+    std::unique_ptr<NumLib::ConvergenceCriterion>&& convergence_criterion,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{process__type}
+    config.checkConfigParameter("type", "SMALL_DEFORMATION");
+    DBUG("Create SmallDeformationProcess.");
+
+    // Process variable.
+    auto process_variables = findProcessVariables(
+        variables, config,
+        {//! \ogs_file_param_special{process__SMALL_DEFORMATION__process_variables__process_variable}
+         "process_variable"});
+
+    DBUG("Associate displacement with process variable \'%s\'.",
+         process_variables.back().get().getName().c_str());
+
+    if (process_variables.back().get().getNumberOfComponents() !=
+        DisplacementDim)
+    {
+        ERR("Wrong number of components");
+        std::abort();
+    }
+
+    // Constitutive relation.
+    // read type;
+    auto const constitutive_relation_config =
+        config.getConfigSubtree("constitutive_relation");
+
+    auto const type =
+        constitutive_relation_config.peekConfigParameter<std::string>("type");
+
+    std::unique_ptr<Solids::SolidsBase<DisplacementDim>> material = nullptr;
+    if (type == "LinearElasticIsotropic")
+    {
+        material = Solids::createLinearElasticIsotropic<DisplacementDim>(
+            parameters, constitutive_relation_config);
+    }
+    else
+    {
+        OGS_FATAL(
+            "Cannot construct constitutive relation of given type \'%s\'.",
+            type.c_str());
+    }
+
+    SmallDeformationProcessData<DisplacementDim> process_data{
+        std::move(material)};
+
+    SecondaryVariableCollection secondary_variables;
+
+    NumLib::NamedFunctionCaller named_function_caller(
+        {"SmallDeformation_displacement"});
```

Comment:
Does that work? I don't expect it, since the displacement is a vectorial quantity.